.github/scripts/end2end/install-kind-dependencies.sh

```diff
@@ -186,22 +222,31 @@ mongodb_wait_for_shards() {
             --eval "db.runCommand({ listshards: 1 }).shards.length"
     )
 
-    [ $count == "1" ]
+    [ $count == "$MONGODB_SHARD_COUNT" ]
 }
 
 mongodb_sharded() {
     local SOLUTION_REGISTRY=metalk8s-registry-from-config.invalid/zenko-base-${VERSION_FULL}
+    local node_count=${NODE_COUNT:-1}
+    local shard_count=${MONGODB_SHARD_COUNT}
 
-    kustomize edit set image \
-        $SOLUTION_REGISTRY/mongodb-sharded=$(get_image_from_deps mongodb-sharded) \
-        $SOLUTION_REGISTRY/os-shell=$(get_image_from_deps mongodb-shell) \
-        $SOLUTION_REGISTRY/mongodb-exporter=$(get_image_from_deps mongodb-sharded-exporter)
+    local base_yaml_path=$(get_mongodb_topology_file $node_count $shard_count)
 
-    kubectl apply -k .
+    sed -i "s|${SOLUTION_REGISTRY}/mongodb-sharded:.*|$(get_image_from_deps mongodb-sharded)|g" "$base_yaml_path"
+    sed -i "s|${SOLUTION_REGISTRY}/os-shell:.*|$(get_image_from_deps mongodb-shell)|g" "$base_yaml_path"
+    sed -i "s|${SOLUTION_REGISTRY}/mongodb-exporter:.*|$(get_image_from_deps mongodb-sharded-exporter)|g" "$base_yaml_path"
 
```

Comment:
why not keep using `kustomize edit set image`, which is more explicit and kind of safer than sed?
if we really need to change, best to use `yq` to edit the required fields...